src/qutip_qip/circuit/mat_renderer.py

```diff
@@ -0,0 +1,1025 @@
+        self._layer_list = {i: [self._start_pad] for i in range(self._qwires)}
+
+        # user defined style
+        style = {} if style is None else style
+        self.style = StyleConfig(**style)
+
+        # fig config
+        self._zorder = {
+            "wire": 1,
+            "wire_label": 1,
+            "gate": 2,
+            "node": 2,
+            "bridge": 2,
+            "connector": 3,
+            "gate_label": 3,
+            "node_label": 3,
+        }
+        self.style.fig_height = (
+            (
+                (self._qwires + self._cwires)
+                * self.style.wire_sep
+                * 0.393701
+                * 3
+            )
+            if self.style.fig_height is None
+            else self.style.fig_height
+        )
+        if self._ax is None:
+            self.fig, self._ax = plt.subplots(
+                figsize=(self.style.fig_width, self.style.fig_height),
+                dpi=self.style.dpi,
+                facecolor=self.style.bgcolor,
+            )
+
+        self.canvas_plot()
+
+    def _get_xskip(self, wire_list: List[int], layer: int) -> float:
+        """
+        Get the xskip (horizontal value for getting to requested layer) for the gate to be plotted.
+
+        Parameters
+        ----------
+        wire_list : list
+            The list of wires the gate is acting on (control and target).
+
+        layer : int
+            The layer the gate is acting on.
+        """
+
+        xskip = []
+        if self.style.align_layer:
+            wire_list = list(range(self._qwires))
+
+        for wire in wire_list:
+            xskip.append(sum(self._layer_list[wire][:layer]))
+
+        return max(xskip)
+
+    def _get_text_width(
+        self,
+        text: str,
+        fontsize: float,
+        fontweight: Union[float, str],
+        fontfamily: str,
+        fontstyle: str,
+    ) -> float:
+        """
+        Get the width of the text to be plotted.
+
+        Parameters
+        ----------
+        text : str
+            The text to be plotted.
+
+        fontsize : float
+            The fontsize of the text.
+
+        fontweight : str or float
+            The fontweight of the text.
+
+        fontfamily : str
+            The fontfamily of the text.
+
+        fontstyle : str
+            The fontstyle of the text.
+        Returns
+        -------
+        float
+            The width of the text in cm.
+        """
+
+        text_obj = plt.Text(
+            0,
+            0,
+            text,
+            fontsize=fontsize,
+            fontweight=fontweight,
+            fontfamily=fontfamily,
+            fontstyle=fontstyle,
+        )
+        self._ax.add_artist(text_obj)
+
+        bbox = text_obj.get_window_extent(
+            renderer=self._ax.figure.canvas.get_renderer()
+        )
+        inv = self._ax.transData.inverted()
+        bbox_data = bbox.transformed(inv)
+        text_obj.remove()
+
+        return bbox_data.width * 2.54 * 3
```

Comment:
What is this 2.54 * 3, what does this magical number mean?

Reply:
`bbox_data.width` returns width in inches, 2.54 is for conversion to centimeters and 3 is multiplied to bring it in same range as of the defined const 😄